tests/NuGetGallery.FunctionalTests.Core/Helpers/ODataHelper.cs

```
@@ -59,17 +59,60 @@ public ODataHelper(ITestOutputHelper testOutputHelper)
             }
         }
 
-        public async Task<bool> ContainsResponseText(string url, params string[] expectedTexts)
+        public async Task<DateTime> GetTimestampOfPackageFromResponse(string url, string propertyName, string packageId, string version = "1.0.0")
         {
-            var request = WebRequest.Create(url);
-            var response = await request.GetResponseAsync().ConfigureAwait(false);
+            WriteLine($"Getting '{propertyName}' timestamp of package '{packageId}' with version '{version}'.");
 
-            string responseText;
-            using (var sr = new StreamReader(response.GetResponseStream()))
+            var packageResponse = await GetPackageDataInResponse(url, packageId, version);
+            if (string.IsNullOrEmpty(packageResponse))
             {
```

`GetPackageDataInResponse` already does this...might be better to throw an exception